buildSrc/src/main/java/org/elasticsearch/gradle/tool/ClasspathUtils.java

```diff
@@ -17,7 +14,8 @@ private ClasspathUtils() {
      *
      * @return if we are currently running in the `elastic/elasticsearch` project
      */
-    public static boolean isElasticsearchProject() {
-        return isElasticsearchProject;
+    public static boolean isElasticsearchProject(Project project) {
```

Comment:
It's now odd to have this method in a class called `ClasspathUtils`. 
We could return `ext.get("isInternal")` if it's set, so tests could set it, and otherwise default to `getResource("/buildSrc.marker") != null` it;s worth to keep all the logic in one place instead of making use of  global build info.

Reply:
Let's keep this as-is for. I have some ideas on making this general pattern less awkward and supporting test-mocking as a first-class thing.